components/ca-treeview.js

```diff
@@ -0,0 +1,267 @@
+    '/components/helpers/create-element.js',
+    '/components/helpers/clear-element.js',
+    '/components/helpers/parent-by-attribute.js',
+    '/components/helpers/cancel-event.js',
+    'document-register-element'
+], (createElement, clearElement, parentByAttribute, cancelEvent) => {
+    /**
+     * Class TreeView for a TreeView web component
+     * @extends HTMLElement
+     */
+    class TreeView extends HTMLElement {
+        /**
+         * createdCallback is called when the component is first created.
+         * creates children, binds events and loads data if src is avaliable.
+         * @returns {undefined}
+         */
+        createdCallback() {
+            this.nav = createElement(this, 'nav', {});
+            this.ol = createElement(this.nav, 'ol');
+            this.nav.onclick = this.eventDelegate.bind(this);
+
+            if (this.dataSrc) {
+                this.loadData();
+            }
+        }
+
+        /**
+         * Called every time an attribute is changed.
+         * @param {string} attrName, the name of the changed attribute.
+         * @param {string} oldVal, previous value associated with attribute.
+         * @param {string} newVal, current value associated with attribute.
+         * @returns {undefined} if data-src or data-root change, then trigger data fetch.
+         */
+        attributeChangedCallback(attrName, oldVal, newVal) {
+            switch (attrName) {
+                case 'data-src': {
+                    if (newVal !== '') {
+                        this.loadData();
+                    }
+                } break;
+
+                case 'data-root': {
+                    this.root = newVal;
+                } break;
+
+                default: break;
+            }
+        }
+
+        /**
+         * Gets the root attribute
+         * @returns {string} data-root attribute, default is empty.
+         */
+        get root() {
+            return this.getAttribute('data-root') || '';
+        }
+
+        /**
+         * Gets the root attribute
+         * @param {string} value, data-root attribute, default is empty.
+         */
+        set root(value) {
+            this.setAttribute('data-root', value || '');
+        }
+
+        /**
+         * Gets the data-src
+         * @returns {string} data-src attribute.
+         */
+        get dataSrc() {
+            return this.getAttribute('data-src');
+        }
+
+        /**
+         * Sets the data-src
+         * @param {string} value, sets the data-src attribute.
+         */
+        set dataSrc(value) {
+            this.setAttribute('data-src', value);
+        }
+
+        /**
+         * Get the data
+         * @returns {array} returns the data.
+         */
+        get data() {
+            return this._data || [];
+        }
+
+        /**
+         * Sets the data
+         * @param {array} value, array to set as the data, defaults to empty. Invokes render.
+         */
+        set data(value = []) {
```

Comment:
same null problem as we had before. i didn't bother making the same comment every time i saw it...